Source/FieldSolver/Fields.H

```diff
@@ -35,6 +36,19 @@ namespace warpx::fields
         Efield_avg_cp,
         Bfield_avg_cp
     };
+
+    inline bool
+    isFieldArray (const FieldType field_type)
+    {
+        return (field_type == FieldType::Efield_aux || field_type == FieldType::Bfield_aux ||
+                field_type == FieldType::Efield_fp  || field_type == FieldType::Bfield_fp ||
+                field_type == FieldType::current_fp || field_type == FieldType::current_fp_nodal ||
+                field_type == FieldType::vector_potential_fp ||
+                field_type == FieldType::Efield_cp  || field_type == FieldType::Bfield_cp ||
+                field_type == FieldType::current_cp ||
+                field_type == FieldType::Efield_avg_fp || field_type == FieldType::Bfield_avg_fp ||
+                field_type == FieldType::Efield_avg_cp || field_type == FieldType::Bfield_avg_cp);
+    }
```

Comment:
```suggestion

        constexpr FieldType ArrayFieldTypes[] = {
            FieldType::Efield_aux, FieldType::Bfield_aux, FieldType::Efield_fp, FieldType::Bfield_fp,
            FieldType::current_fp, FieldType::current_fp_nodal, FieldType::vector_potential_fp,
            FieldType::Efield_cp, FieldType::Bfield_cp, FieldType::current_cp,
            FieldType::Efield_avg_fp, FieldType::Bfield_avg_fp, FieldType::Efield_avg_cp, FieldType::Bfield_avg_cp};

    inline bool
    isFieldArray (const FieldType field_type)
    {
        return std::find(
            std::begin(ArrayFieldTypes), std::end(ArrayFieldTypes),
            field_type) != std::end(ArrayFieldTypes);
    }
```
I would suggest this optional change in order to improve readability.

Reply:
Thanks for the suggestion. I ran into compile issues with this usage of std::find() that I couldn't readily figure out. I settled on a slightly different version that does not use std::find(). Let me know what you think.

`constexpr bool
   isFieldArray (const FieldType field_type)
   {
        for (const auto& f : ArrayFieldTypes) {
            if (f == field_type) { return true; }
        }
        return false;
 }`

Reply:
Clang tidy complained. **error: replace loop by 'std::any_of()'**

It wants me to replace the for loop with std::any_of().

Reply:
Yes,  the reason is that we enable this check in clang-tidy:

- [readability/use-anyofallof](https://releases.llvm.org/15.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/readability/use-anyofallof.html)

In this case I personally think that the final solution is quite elegant!

However, in general, if you find some `clang-tidy` suggestions too pedantic, please don't hesitate to add a point in the agenda of the developers' meeting: we may decide to remove a check or, in some cases, to tune how the check works.

Reply:
I think the clang-tidy suggestion here is good.